winch/codegen/src/isa/x64/asm.rs

```diff
@@ -168,6 +168,18 @@ impl From<OperandSize> for Option<ExtMode> {
     }
 }
 
+impl SplatKind {
+    /// Get the operand size for `vpbroadcast`.
+    pub(super) fn vpbroadcast_operand_size(&self) -> OperandSize {
```

Comment:
Instead of making this specific to x64, can we instead call this `lane_size` or something along those lines and move the implementation to `masm.rs`? I don't think we'd only need this for x64.